src/lib/graphql.tsx

```diff
@@ -0,0 +1,66 @@
+import { gql } from "@apollo/client";
+
+export const GET_USERS = gql`
+  query getUser {
+    users {
+      id
+      email
+      username
+      firstName
+      lastName
+      dateJoined
+      lastOnline
+      online
+      created_at
+      updated_at
+      avatar
+      connectedWalletAddress
+      country
+      timezone
+      newsletter
+      journeyUpdates
+      learningJourneys {
+        id
+        title
+        created_at
+        updated_at
+      }
+    }
+  }
+`;
+
+export const UPDATE_SETTINGS = gql`
+  mutation updateNotificationSettings(
+    $id: String!
+    $newsletter: Boolean
+    $journeyUpdates: Boolean
+    $connectedWalletAddress: String
+    $username: String
```

Comment:
you are passing query variable "username" is missing in $set?

Reply:
My thinking on this was to create a general update_Settings mutation that takes all settings variables. This way in the future when we make fields editable we can continue to use the same query.

I looked on the docs and it said if you don't pass a nullable variable it just doesn't update, which is what we'd want.

Let me know your thoughts.

Reply:
If it is intentional then I am ok, I thought you missed updating the mutation query :)
Can you please confirm your changes are working, I mean when clicking the save button there is no error?

Reply:
You were right. Needed to add username to $set :)